resources/user.py

```diff
@@ -36,25 +36,16 @@ def post(self):
 class User(Resource):
     @admin_required
     def get(self, user_id):
-        user = UserModel.find_by_id(user_id)
-
-        if not user:
-            return {"message": "User not found"}, 404
+        user = UserModel.find(user_id)
 
         user_info = user.json()
 
         if user.role == RoleEnum.PROPERTY_MANAGER:
-            user_info["properties"], tenant_list = zip(
-                *(
-                    (p.json(), p.tenants)
-                    for p in PropertyModel.find_by_manager(user_id)
-                    if p
-                )
-            )
+            properties_list = PropertyModel.find_by_manager(user_id)
+            user_info["properties"] = [p.json() for p in properties_list]
 
-            tenant_IDs = [tenant.id for sublist in tenant_list for tenant in sublist]
-            tenants_list = [TenantModel.find_by_id(t) for t in set(tenant_IDs)]
-            user_info["tenants"] = [t.json() for t in tenants_list if t]
+            tenants_json_2D = [p.tenants() for p in properties_list]
+            user_info["tenants"] = list(chain.from_iterable(tenants_json_2D))
```

Comment:
Can we move this logic to the serializer? Also it seems like we are missing a relationship here. I don't think it should be this complex to serialized some data.

Reply:
In hindsight, this may be better encapsulated as a method in the tenants model. Find tenants by property. 
I wonder about your comment about a missing relationship. Tenant and Lease have a relationship. Lease contains a key for property. A new relationship would have to be between Tenant and Property. Is this possible? It seems redundant?